pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -297,11 +299,16 @@ def _register_ddp_hooks(self) -> None:
                 ddp_comm_wrapper=self._ddp_comm_wrapper,
             )
 
-    def configure_ddp(self):
-        self.pre_configure_ddp()
+    def configure_ddp(self) -> None:
+        # skip wrapping the model if we are not fitting as no gradients need to be exchanged
+        trainer_fn = self.lightning_module.trainer.state.fn
+        if trainer_fn != TrainerFn.FITTING:
+            rank_zero_debug(f"In {trainer_fn} stage: Skipping wrapping the model with DistributedDataParallel")
+            return
         self._model = DistributedDataParallel(
             LightningDistributedModule(self.model), device_ids=self.determine_ddp_device_ids(), **self._ddp_kwargs
         )
+        self.pre_configure_ddp()
```

Comment:
why has pre_configure moved after configuring ddp?
This must be a bug/unintentional. Let's move it back to before wrapping and after your newly introduced if statement? same applies to the other plugins.

Reply:
@ninginthecloud this needs to be flipped

Reply:
yep. you are right. I updated pre_configure_ddp in ddp and ddp_spawn.